relay.go

```diff
@@ -654,6 +664,80 @@ func (r *Relayer) handleLocalCallReq(cr *lazyCallReq) bool {
 	return true
 }
 
+func (r *Relayer) fragmentingSend(f *lazyCallReq, relayToDest relayItem, origID uint32) error {
+	fragWriter := newFragmentingWriter(
+		NullLogger, r.newFragmentSender(relayToDest.destination, f, origID),
+		f.checksumType.New(),
+	)
+	defer r.conn.opts.FramePool.Release(f.Frame)
+
+	arg2Writer, err := fragWriter.ArgWriter(f.isArg2Fragmented)
+	if err != nil {
+		return fmt.Errorf("get arg2 writer: %v", err)
+	}
+
+	if err := writeArg2WithAppends(arg2Writer, f.arg2(), f.arg2appends); err != nil {
+		return fmt.Errorf("write arg2: %v", err)
+	}
+	if f.isArg2Fragmented {
+		return nil
+	}
+
+	if err := NewArgWriter(fragWriter.ArgWriter(true)).Write(f.arg3()); err != nil {
+		return errors.New("arg3 write failed")
+	}
+
+	return nil
+}
+
+func writeArg2WithAppends(w io.WriteCloser, arg2 []byte, appends []keyVal) (err error) {
+	defer w.Close()
+
+	var nh uint16
+	if len(arg2) > 0 {
+		nh = binary.BigEndian.Uint16(arg2[:2])
+	}
+	nh += uint16(len(appends))
+	if err := writeUint16(w, nh); err != nil {
+		return fmt.Errorf("write arg2 nh: %v", err)
+	}
+
+	if len(arg2) > 0 {
+		if _, err := w.Write(arg2[2:]); err != nil {
+			return fmt.Errorf("write arg2: %v", err)
+		}
+	}
+	for _, kv := range appends {
+		if err := writeLen16Data(w, kv.key); err != nil {
+			return fmt.Errorf("append arg2 key: %v", err)
+		}
+		if err := writeLen16Data(w, kv.val); err != nil {
+			return fmt.Errorf("append arg2 val: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func writeLen16Data(w io.Writer, b []byte) error {
+	if err := writeUint16(w, uint16(len(b))); err != nil {
+		return fmt.Errorf("write data length: %v", err)
+	}
+	if _, err := w.Write(b); err != nil {
+		return fmt.Errorf("write data: %v", err)
+	}
+	return nil
+}
+
+func writeUint16(w io.Writer, n uint16) error {
+	var sizeBuf [2]byte
+	binary.BigEndian.PutUint16(sizeBuf[:], n)
+	if _, err := w.Write(sizeBuf[:]); err != nil {
+		return err
+	}
+	return nil
```

Comment:
if we're not wrapping the error, can simplify to:

```suggestion
	_, err := w.Write(sizeBuf[:]); err != nil {
	return err
```